grammar.js

```diff
@@ -84,6 +89,21 @@ module.exports = grammar(C, {
       alias($.operator_cast_declaration, $.declaration),
     ),
 
+    _block_item: ($, original) => choice(
+      original,
+      $.namespace_definition,
+      $.concept_definition,
+      $.namespace_alias_definition,
+      $.using_declaration,
+      $.alias_declaration,
+      $.static_assert_declaration,
+      $.template_declaration,
+      $.template_instantiation,
+      alias($.constructor_or_destructor_definition, $.function_definition),
+      alias($.operator_cast_definition, $.function_definition),
+      alias($.operator_cast_declaration, $.declaration),
+    ),
```

Comment:
What is this for?

Reply:
reworked top level items in C, this just adds certain bits missing to _block_item that _top_level_item had

Reply:
Is it possible to consolidate all these various lists of "top level-like items? It seems like there's quite a bit of copy/paste and is likely one of the reasons for the pretty large increase in state count and parser size this commit introduced.

Reply:
Not exactly, because top level items and block items use different versions of expression statements to keep top level declarations less conflicting with binary expressions. 
https://github.com/tree-sitter/tree-sitter-c/pulls?q=is%3Apr+is%3Aclosed